src/Coordination/KeeperServer.cpp

```diff
@@ -932,4 +932,25 @@ KeeperLogInfo KeeperServer::getKeeperLogInfo()
     return log_info;
 }
 
+bool KeeperServer::requestLeader()
+{
+    if (isLeader())
+        return true;
+
+    raft_instance->request_leadership();
+
+    auto timeout = coordination_settings->election_timeout_upper_bound_ms.totalMilliseconds() == 0 ? 2000 :
+        coordination_settings->election_timeout_upper_bound_ms.totalMilliseconds();
+
+    Stopwatch watch;
+
+    /// Pull wait to become leader
+    while (!isLeader() && watch.elapsedMilliseconds() < static_cast<uint64_t>(timeout))
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+    }
+
+    return isLeader();
```

Comment:
I meant polling in the tests, not here.
I think it was better before when it was async, so we can leave it to the user to decide how much to wait.
But the first check is good, if it's already leader this command should return true and not send the request (`request_leadership` returns `false` if it's already leader).
WDYT?

Reply:
ok